src/main/java/mate/academy/spring/boot/service/BookServiceImpl.java

```diff
@@ -50,7 +63,19 @@ public void deleteById(Long id) {
     }
 
     @Override
-    public List<BookDto> search(BookSearchParametersDto searchParameters) {
-        return Collections.emptyList();
+    public List<BookDto> search(BookSearchParametersDto params, Pageable pageable) {
+        Specification<Book> bookSpecification = bookSpecificationBuilder.build(params);
+        return bookRepository.findAll(bookSpecification, pageable)
+                .stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    private String generateUniqueIsbn() {
```

Comment:
You can provide validation for isbn by using pattern and validator. Don't need to generate it.